server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -284,6 +284,9 @@ func (ds *cachedMysql) TeamMDMConfig(ctx context.Context, teamID uint) (*fleet.T
 	if err != nil {
 		return nil, err
 	}
+
+	ds.c.Set(key, cfg, ds.teamMDMConfigExp)
+
```

Comment:
this is the caching issue found by @lucasmrod

Reply:
Hmm I investigated that cache key a bit and we should delay adding it (or make the necessary missing changes).

Currently the expiration is never set AFAICT, so it's expiration is 0 (looking at the cache's implementation, it looks like it would default to 5 minutes which is probably too long (other team stuff is 1m) but in any case we should explicitly set its expiration like we do for other keys).

More importantly, it looks like there's no `Clone` implementation on `*fleet.TeamMDM` (the `cached_mysql.cloner` interface), which means it will be cloned by reflection which caused major performance issues at scale in the past.

Reply:
hmmm, yeah you're right, this clearly needs more thought.

this was a "nice to include" that's not related to the actual issue, so I will create a separate issue with your notes and revert this.

Reply:
for future reference, this is the issue: https://github.com/fleetdm/fleet/issues/14708